assets/js/callback.js

```diff
@@ -1,24 +1,15 @@
-function redirectUser() {
-  location.href = getReferringPage() ?? "https://www.tpen.org/interfaces"
-}
-
-function getReferringPage() {
-  try {
-    const base64Hash = location.hash.split("state=")[1].split("&")[0]
-    const decodedUrl = b64toUrl(base64Hash)
-    return decodedUrl
-  } catch (err) {
-    return false
-  }
-}
+window.onload = redirectUser()
 
 /**
- * Follows the 'base64url' rules to decode a string.
- * @param {String} base64str from `state` parameter in the hash from Auth0
- * @returns referring URL
+ *  Detect and get the value of returnTo from the origin address /callback/?returnTo=
+ *  If there is no returnTo, default to the origin homepage.
+ *  Note that performLogout() in auth.js will always add a returnTo, and it may just be the origin homepage.
  */
-function b64toUrl(base64str) {
```

Comment:
replacing this means we cannot ever come here with a base64hash, which maybe is fine.